### PR TITLE
[EOS-21401] : S3server reset and cleanup phase should be independent of template/config files

### PR DESCRIPTION
**Pull request by pranavPawar07**
Signed-off-by: Pranav Diliprao Pawar <pranav.pawar@seagate.com>

### CODE REVIEW

File: scripts/provisioning/setupcmd.py [hunk 1]
```diff
@@ -538,4 +565,42 @@ def DeleteFileOrDirWithRegex(self, path: str, regex: str):
             shutil.rmtree(file)
         except Exception as e:
           self.logger.error(f'ERROR: DeleteFileOrDirWithRegex(): Failed to delete: {file}, error: {str(e)}')
-          raise e
\ No newline at end of file
+          raise e
+
+  def get_iam_admin_credentials(self):
+    """Used for reset and cleanup phase to get the iam-admin user and decrypted passwd."""
+    op_file = "/opt/seagate/cortx/auth/resources/authserver.properties"
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:572]: Let us not hardcode files and locations inside function

File: scripts/provisioning/setupcmd.py [hunk 2]
```diff
@@ -538,4 +565,42 @@ def DeleteFileOrDirWithRegex(self, path: str, regex: str):
...
+    # self.ldap_passwd = opfileconfstore.get_config('ldapLoginPW')
+    enc_ldap_passwd = opfileconfstore.get_config('ldapLoginPW')
+    cipher_key = s3cipher_obj.generate_key()
+
+    if enc_ldap_passwd != None:
+      self.ldap_passwd = s3cipher_obj.decrypt(cipher_key, enc_ldap_passwd)
+
+  def get_ldap_root_credentials(self):
+    """Used for reset and cleanup phase to get the ldap root user and decrypted passwd."""
+    op_file = "/opt/seagate/cortx/s3/s3backgrounddelete/s3_cluster.yaml"
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:589]: Let us not hardcode files and locations inside function

File: scripts/provisioning/setupcmd.py [hunk 3]
```diff
@@ -538,4 +565,42 @@ def DeleteFileOrDirWithRegex(self, path: str, regex: str):
...
+  def get_iam_admin_credentials(self):
+    """Used for reset and cleanup phase to get the iam-admin user and decrypted passwd."""
+    op_file = "/opt/seagate/cortx/auth/resources/authserver.properties"
+    opfileconfstore = S3CortxConfStore(f'properties://{op_file}', 'read_ldap_idx')
+    s3cipher_obj = CortxS3Cipher(None,
+                              False,
+                              0,
+                              self.get_confkey('CONFSTORE_OPENLDAP_CONST_KEY'))
+
+    self.ldap_user = "sgiamadmin"
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:579]: Make this a constant declaration as well

**Comment by nileshgovande** [comment]
Add entries to s3backgrounddelete/s3backgrounddelete/config/s3_cluster_unsafe_attributes.yaml

File: scripts/provisioning/resetcmd.py [hunk 4]
```diff
@@ -133,13 +136,15 @@ def purge_messages(self, producer_id: str, msg_type: str, delivery_mechanism: st
 
   def DeleteLdapAccountsUsers(self):
     """Deletes all LDAP data entries e.g. accounts, users, access keys using admin credentials."""
-    self.read_ldap_credentials()
+    # self.read_ldap_credentials()
 
     try:
+      print("rootdnuser : ", self.ldap_root_user)
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:142]: Remove prints

File: scripts/provisioning/resetcmd.py [hunk 5]
```diff
@@ -152,6 +157,9 @@ def DeleteLdapAccountsUsers(self):
                                   'const_cipher_secret_str': "s3backgroundsecretkey",
                                   'const_cipher_access_str': "s3backgroundaccesskey"
                                 }
+
+      print("user : ", self.ldap_user)
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:161]: Remove prints

File: scripts/provisioning/setupcmd.py [hunk 6]
```diff
@@ -130,18 +131,44 @@ def read_ldap_credentials(self):
 
       encrypted_ldapadmin_pass = self.get_confvalue(self.get_confkey('CONFIG>CONFSTORE_LDAPADMIN_PASSWD_KEY'))
 
+      if encrypted_ldapadmin_pass != None:
+        self.ldap_passwd = s3cipher_obj.decrypt(cipher_key, encrypted_ldapadmin_pass)
+
+      if self.ldap_user != "sgiamadmin":
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:137]: The validator should be in post_install phase and all the phases

File: scripts/provisioning/resetcmd.py [hunk 7]
```diff
@@ -133,13 +136,15 @@ def purge_messages(self, producer_id: str, msg_type: str, delivery_mechanism: st
 
   def DeleteLdapAccountsUsers(self):
     """Deletes all LDAP data entries e.g. accounts, users, access keys using admin credentials."""
-    self.read_ldap_credentials()
+    # self.read_ldap_credentials()
```
Comment (sachitanands) [inline scripts/provisioning/resetcmd.py:139]: please remove this method if its not in use anymore

File: scripts/provisioning/setupcmd.py [hunk 8]
```diff
@@ -209,6 +209,9 @@ def phase_prereqs_validate(self, phase_name: str):
     if not os.path.isfile(self._preqs_conf_file):
       raise FileNotFoundError(f'pre-requisite json file: {self._preqs_conf_file} not found')
     _prereqs_confstore = S3CortxConfStore(f'json://{self._preqs_conf_file}', f'{phase_name}')
+
+    if self.ldap_user != "sgiamadmin":
+      raise ValueError('provide correct ldap username')
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:214]: Fix this message for more clearity

**Comment by pranavPawar07** [comment]
Mini prov single node **(SUCCESS)** : http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-s3server/1433/console

**Review by nileshgovande — APPROVED** [review]

**Review by vimalraghwani20 — APPROVED** [review]

**Pull request merged**